eclair-core/src/main/scala/fr/acinq/eclair/wire/protocol/LightningMessageTypes.scala

```diff
@@ -312,16 +312,22 @@ object NodeAddress {
   /**
    * Creates a NodeAddress from a host and port.
    *
-   * Note that non-onion hosts will be resolved.
+   * Note that only IP v4 and v6 hosts will be resolved, onion and DNS hosts names will not be resolved.
    *
    * We don't attempt to resolve onion addresses (it will be done by the tor proxy), so we just recognize them based on
    * the .onion TLD and rely on their length to separate v2/v3.
+   *
+   * We resolve host names comprised of only numbers and periods (IPv4) or that contain a colon (IPv6).
+   * Other host names are assumed to be a DNS name and are not immediately resolved.
+   *
    */
   def fromParts(host: String, port: Int): Try[NodeAddress] = Try {
+    val ipv4v6 = "^([0-9.]*)?$|(:)".r
```

Comment:
Isn't there a function in the networking utilities we have available on the JVM to do that?

Reply:
fixed in [fd32764](https://github.com/ACINQ/eclair/pull/2234/commits/fd32764cf3f01be1f6e35832f365250de252c2aa)

Reply:
Updated the comment too in [76d24d2](https://github.com/ACINQ/eclair/pull/2234/commits/76d24d2513bddcb0fd7de5930bbb32b4198d0005)